### PR TITLE
Add max background jobs to configuration

### LINKED ISSUE #1292: Enable max_background_jobs
### Discussed in https://github.com/apache/incubator-kvrocks/discussions/1284

<div type='discussions-op-text'>

<sup>Originally posted by **xiaobiaozhao** February 27, 2023</sup>
Use `max_background_jobs` instead of `max_background_compactions` and `max_background_flushes`

https://github.com/facebook/rocksdb/blob/main/include/rocksdb/options.h#L699</div>

### PR DESCRIPTION
**Pull request by bruceqiwang**

### CODE REVIEW

File: kvrocks.conf [hunk 1]
```diff
@@ -575,15 +575,8 @@ rocksdb.target_file_size_base 128
...
 
-# Maximum number of concurrent background compaction jobs, submitted to
-# the default LOW priority thread pool.
-rocksdb.max_background_compactions 4
-
-# Maximum number of concurrent background memtable flush jobs, submitted by
-# default to the HIGH priority thread pool. If the HIGH priority thread pool
-# is configured to have zero threads, flush jobs will share the LOW priority
-# thread pool with compaction jobs.
-rocksdb.max_background_flushes 4
```
Comment (PragmaTwice) [inline kvrocks.conf:586]: AFAIK it is a BREAKING CHANGE, as you remove some configuration options in kvrocks.conf. You may need to explain more about this.

Comment (git-hulk) [reply kvrocks.conf:586]: Yes, I think we can mark them as deprecated first and set the default value to -1, then encourage users to use the max_background_jobs like RocksDB: https://github.com/facebook/rocksdb/blob/8c445407b79ab02620e15cdd1257b1c99ea09601/include/rocksdb/options.h#L701

File: src/config/config.cc [hunk 2]
```diff
@@ -206,6 +204,7 @@ Config::Config() {
        new IntField(&RocksDB.max_bytes_for_level_multiplier, 10, 1, 100)},
       {"rocksdb.level_compaction_dynamic_level_bytes", false,
        new YesNoField(&RocksDB.level_compaction_dynamic_level_bytes, false)},
+      {"rocksdb.max_background_jobs", false, new IntField(&RocksDB.max_background_jobs, 2, 0, 32)},
```
Comment (torwig) [inline src/config/config.cc:207]: I guess the default value for `max_background_jobs` should be 4, not 2. Right?

**Comment by bruceqiwang** [comment]
report a new pr

**Comment by bruceqiwang** [comment]
report a new pr and close this one

**Comment by xiaobiaozhao** [comment]
close this pr, new pr is  https://github.com/apache/incubator-kvrocks/pull/1374

**Pull request closed without merging**